backend/test/v2/integration/pipeline_version_api_test.go

```diff
@@ -301,12 +301,13 @@ func (s *PipelineVersionApiTest) TestV2Spec() {
 	test.DeleteAllPipelines(s.pipelineClient, t)
```

Comment:
@rimolive it may be worth updating `test.DeleteAllPipelines` so that it doesn't return until the API server says all the pipelines are actually deleted (i.e. the cache is updated). I think that would fix any test flakes from race conditions.